source/node/design_docs/interaction_alice_node_sdk.rst

```diff
@@ -0,0 +1,147 @@
+  
+Once both the parties agree on the initial state of the channel and fund the
+channel on the blockchain, it progress to the transact phase.
+
+Transact Phase
+--------------
+
+In this phase, a user can either propose a new state update or respond to an
+incoming state update by either accepting or rejecting it.
+
+.. image:: ../../_generated/node/seq_2_transact.svg
+  :align: Center
+  :alt: Image not available
+
+
+.. note::
+  UpdateResponders is a data field to track all the Responder objects for
+  incoming channel updates, as the responder is used only when the user sends a
+  Respond request.
+
+.. note::
+  Incoming update notification are cached if there is no active subscription.
+
+  Entries from the notification cache are removed only when they have been
+  delivered to a valid subsctiption or when the session or channel is closed.
+  
+
+The channel participants can send as many channel updates as they desire. When
+any one of the channel participants want to close the channel, they send a
+channel close request to the perun-node and the channel progresses to the
+register phase.
+
+.. note::
+  Collaborative and Non Collaborative Close:
+
+  When a close request is sent by any of the channel participants, perun-node
+  will mark the latest state of the channel as final and send an update request
+  to the peer.  Like any other channel update, this can be accepted / rejected
+  as described in "Transact phase". If it is
+
+  - accepted by the peer -> Collaborative close will follow. 
+  - rejected by the peer -> Non Collaborative close will follow.
+
+  In both cases, the node will call `channel.Register` in the next phase. The
+  difference is that,
+
+  1. In case of collaborative close, the channel state will be directly
+     concluded on the blockchain, without needing to register and wait for the
+     challenge duration to expire.
+
+  2. In case of non-collaborative close, the channel state will first have to
+     be registered on the blockchain. Both the participants have to then wait
+     for the challenge duration to pass and then conclude it. The challenge
+     duration allows time for the other pariticipant to refuse if an older
+     state was registered.
+     
+  Collaborative Close is an optimization to close the channel with lesser
+  transactions on the blockchain (one instead of two) and with no waiting
+  period.
+
+
+Register Phase
+--------------
+
+In this phase, either a non-final state is registered or a final state is
+directly concluded on the blockchain. This depends on if the pariticipant have
+chosen non-collaborative or collaborative close respectively.
+
+.. image:: ../../_generated/node/seq_3_register.svg
+  :align: Center
+  :alt: Image not available
+
+If a non-final state was registered on the blockchain and other participant
+sees that it is not the latest state, it can refute to the registered state by
+registering the latest state to the blockchain. This can go on until the no
```

Comment:
```suggestion
registering the latest state to the blockchain. This can go on until no
```

Reply:
igned-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>